consensus/polybft/consensus_runtime.go

```diff
@@ -205,7 +198,9 @@ func (c *consensusRuntime) OnBlockInserted(block *types.Block) {
 		c.logger.Error("on block inserted error", "err", err)
 	}
 
-	if c.isEndOfEpoch(block.Header.Number) {
+	// TODO - this condition will need to be changed to recognize that either slashing happened
+	// or epoch reached its fixed size
```

Comment:
Why it that different?

Reply:
`OnBlockInserted` is called on block insert through `consensus` and through `syncer` so we can not just check if fixed size of epoch is met. Maybe this was the epoch ending block because we had slashing. We would need to persist some flag that slashing occurred and check it here. It won't be a big problem to do that, but we need the slashing implemented both on client and on smart contract.